### PR TITLE
Feat/brand colors figma variables

### LINKED ISSUE #660: Update CSS-in-JS with Brand Color Tokens - Light Theme

### PR DESCRIPTION
**Pull request by garrettbear**
## **Description**
Update to start using figma color variables for `brandColor`, `lightTheme`, and `darkTheme`. This involves a new json architect from Figma. 

> This update does include some breaking changes
- Network Colors Removed
- Colors previously noted as `Deprecated` have officially been removed
- Changes to some color shades (detailed notes will be found in migration doc) 

## **Related issues**

Fixes: #660  #660 

## **Manual testing steps**

1. Run Storybook
2. View Brand Colors  
3. View Light Theme 
4. View Dark Theme 

### **Before**

https://github.com/MetaMask/design-tokens/assets/8112138/c5052e7f-7447-446e-bd94-a484c3e0bf19

### **After**

https://github.com/MetaMask/design-tokens/assets/26469696/969f93d2-b95e-4f08-b778-56e816b38218

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by garrettbear** [comment]
@brianacnguyen @georgewrmarshall I noticed we need to add `white` and `black` to brand colors for css

File: MIGRATION.md [hunk 1]
```diff
@@ -29,3 +30,113 @@ import '../../node_modules/@metamask/design-tokens/dist/styles.css';
...
+- secondary colors removed
+- error-shadow removed
+- error-disabled removed
+- warning-alternative removed
+- warning-disabled removed
+- success-alternative removed
+- success-disabled removed
+- info-alternative removed
+- info-disabled removed
+- network colors removed
```
Comment (georgewrmarshall) [inline MIGRATION.md:123]: I think it would help if we are specific about exactly what JS and CSS tokens have been removed

File: MIGRATION.md [hunk 2]
```diff
@@ -29,3 +30,113 @@ import '../../node_modules/@metamask/design-tokens/dist/styles.css';
...
+### Brand Colors
+
+- Grey updates
+- grey000 added
+- grey025 added
+- grey050 added
+- grey030 removed
+- grey040 removed
+- grey750 removed
+- grey1000 added
```
Comment (georgewrmarshall) [inline MIGRATION.md:45]: I suggest that once we have all the changes, we categorize them according to the [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) documentation: Added, Changed, and Removed. This will help with checking for breaking changes in each client codebase. While additions and changes might be less critical, it is important to identify what has been removed and how to resolve those instances.

File: automation-scripts/brandColor.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
+const fs = require('fs').promises;
```
Comment (georgewrmarshall) [inline automation-scripts/brandColor.js:1]: Can we add comments at the top of these files explaining that they aren't used in the build steps and that there is still some manual steps to transpile the JSON to JS tokens

File: MIGRATION.md (hunk 1 above)
Comment (georgewrmarshall) [inline MIGRATION.md:123]: I think it would help if we are specific about exactly what JS and CSS tokens have been removed

File: automation-scripts/themeColors.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
+const fs = require('fs').promises;
```
Comment (georgewrmarshall) [inline automation-scripts/themeColors.js:1]: Same here

File: docs/BrandColors.stories.tsx [hunk 5]
```diff
@@ -51,7 +51,7 @@ export const CSS: Story = {
   },
 };
 
-export const JS: Story = {
+export const JS = {
```
Comment (georgewrmarshall) [inline docs/BrandColors.stories.tsx:54]: Is this change necessary? The other Stories are typed can we either remove it from all stories or add it here?
```suggestion
export const JS: Story = {
```

File: docs/BrandColors.stories.tsx [hunk 6]
```diff
@@ -1,9 +1,9 @@
 import React from 'react';
 import type { Meta, StoryObj } from '@storybook/react';
-import { brandColor as brandColorJS } from '../src/js';
```
Comment (georgewrmarshall) [inline docs/BrandColors.stories.tsx:3]: I was getting a clash with the `brandColor` object from the useJsonColor hook which is why it is aliased here are you not?

File: docs/ThemeColors.stories.tsx [hunk 7]
```diff
@@ -41,12 +38,7 @@ export const FigmaDarkTheme = {
           padding: '1rem',
         }}
       >
-        <ColorSwatchGroup
-          swatchData={darkTheme}
-          borderColor={darkTheme.border.muted.value}
-          textBackgroundColor={darkTheme.background.default.value}
-          textColor={darkTheme.text.default.value}
-        />
+        <ColorSwatchGroup swatchData={darkTheme} theme="#24272a" />
```
Comment (georgewrmarshall) [inline docs/ThemeColors.stories.tsx:41]: Can we avoid using static hex values outside of the tokens themselves?

File: docs/components/ColorSwatchGroup/deprecated/ColorSwatchGroup.tsx [hunk 8]
```diff
@@ -0,0 +1,121 @@
+import React, { FunctionComponent } from 'react';
+import { Theme } from '../../../utils/useJsonColor';
+import { ColorSwatch } from '../..';
```
Comment (georgewrmarshall) [inline docs/components/ColorSwatchGroup/deprecated/ColorSwatchGroup.tsx:3]: Is this being used or can we remove it?

File: docs/utils/useJsonColor.ts [hunk 9]
```diff
@@ -23,24 +23,18 @@ type CompiledColors = {
...
 
   useEffect(() => {
-    /**
-     * Parses a referenced color value and resolves it based on the current theme.
-     * If the value is a reference (enclosed in curly braces), it navigates through the theme object.
-     *
-     * @param value - The color value or reference to resolve.
-     * @param theme - The theme object to resolve references against.
-     * @returns The resolved color value.
-     */
```
Comment (georgewrmarshall) [inline docs/utils/useJsonColor.ts:43]: What's the reason for removing these comments?

File: docs/utils/useJsonColor.ts [hunk 10]
```diff
@@ -56,36 +50,54 @@ export const useJsonColor = (): CompiledColors => {
       return value; // Return the direct value if not a reference.
     };
 
-    /**
-     * Compiles color themes from provided JSON theme definitions.
-     * Each color value is checked and potentially resolved using parseColorValue.
-     *
-     * @param themes - Object containing various theme definitions.
-     * @returns Object containing compiled and resolved themes.
-     */
```
Comment (georgewrmarshall) [inline docs/utils/useJsonColor.ts:65]: Any reason to remove these comments?

File: docs/utils/useJsonColor.ts [hunk 11]
```diff
@@ -56,36 +50,54 @@ export const useJsonColor = (): CompiledColors => {
...
+                  (description ?? '') +
+                  (value === resolvedValue ? '' : ` (${value})`),
+              };
+            });
+          }
         });
         compiledColors[themeName] = tempThemeColors;
       });
+
       return compiledColors;
```
Comment (georgewrmarshall) [inline docs/utils/useJsonColor.ts:101]: Will this file conflict with the updates here https://github.com/MetaMask/design-tokens/pull/699

File: src/css/brand-colors.css [hunk 12]
```diff
@@ -1,73 +1,116 @@
...
  * Instead use the "--color-" variables to ensure
  * theme compatible styles
  */
+
 :root {
-  --brand-colors-white-white000: #ffffff;
-  --brand-colors-white-white010: #fcfcfc;
-  --brand-colors-black-black000: #000000;
-  --brand-colors-grey-grey030: #fafbfc;
-  --brand-colors-grey-grey040: #f2f4f6;
```
Comment (georgewrmarshall) [inline src/css/brand-colors.css:13]: Can we add all these removals to the migration docs?

File: src/css/dark-theme-colors.css [hunk 13]
```diff
@@ -1,65 +1,151 @@
...
+  --color-primary-muted: var(--brand-colors-blue-blue400-10);
+  /* For elements used on top of primary/default. Used for text, icon or border */
   --color-primary-inverse: var(--brand-colors-grey-grey900);
-  --color-primary-disabled: #1098fc80;
-  --color-primary-shadow: #0376c933;
-  --color-secondary-default: var(--brand-colors-orange-orange400);
-  --color-secondary-alternative: var(--brand-colors-orange-orange300);
-  --color-secondary-muted: #f8883b26;
-  --color-secondary-inverse: var(--brand-colors-grey-grey900);
-  --color-secondary-disabled: #f8883b80;
```
Comment (georgewrmarshall) [inline src/css/dark-theme-colors.css:37]: Same for all these

File: src/css/light-theme-colors.css [hunk 14]
```diff
@@ -1,81 +1,152 @@
...
   --color-primary-alternative: var(--brand-colors-blue-blue600);
-  --color-primary-muted: #0376c919;
-  --color-primary-inverse: var(--brand-colors-white-white000);
-  --color-primary-disabled: #0376c980;
-  --color-primary-shadow: #0376c933;
-  --color-secondary-default: var(--brand-colors-orange-orange500);
-  --color-secondary-alternative: var(--brand-colors-orange-orange600);
-  --color-secondary-muted: #f66a0a19;
-  --color-secondary-inverse: var(--brand-colors-white-white000);
-  --color-secondary-disabled: #f66a0a80;
```
Comment (georgewrmarshall) [inline src/css/light-theme-colors.css:38]: Add to migration docs

File: src/css/typography.css [hunk 15]
```diff
@@ -1,7 +1,9 @@
 /**
  * Typography
  */
+
 :root {
+  /* typography */
```
Comment (georgewrmarshall) [inline src/css/typography.css:6]: nit: Can we revert these changes to keep code changes down?

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Looking really good! Can we ensure that all Added, Changed and Removed CSS and JS tokens are listed in the migration docs. Also looks like there is an issue with displaying Figma tokens

Screenshot 2024-05-23 at 12 29 56 PM
Screenshot 2024-05-23 at 12 30 02 PM

Something like 
### Added

#### CSS
```
--brand-colors-grey-grey000
--brand-colors-grey-grey025
--brand-colors-grey-grey050
--brand-colors-grey-grey1000
--brand-colors-blue-blue025
--brand-colors-blue-blue050
// etc
```

#### JS 
```
brandColor.grey000
brandColor.grey025
brandColor.grey050
// 
```

### Removed

#### CSS
```
--brand-colors-grey-grey030
--brand-colors-grey-grey040
--brand-colors-grey-grey750
--brand-colors-blue-blue000
--brand-colors-green-green000
--brand-colors-red-red000
// etc
```

#### JS 
```
brandColor.grey030
brandColor.grey040
brandColor.grey750
brandColor.blue000
brandColor.green000
brandColor.red000
// etc

lightTheme.colors.overlay.inverse
lightTheme.colors.primary.shadow
lightTheme.colors.primary.disabled
lightTheme.colors.secondary
lightTheme.colors.error.shadow
// etc
```

### Changed

#### CSS
```
--brand-colors-white-white000 to --brand-colors-white
--brand-colors-black-black000 to --brand-colors-black
```

#### JS
```
brandColor.white000 to brandColor.white
brandColor.black000 to `brandColor.black
```

File: docs/BrandColors.stories.tsx (hunk 6 above)
Comment (garrettbear) [reply docs/BrandColors.stories.tsx:3]: not getting any issue, but if you are getting a clash then I will update

File: MIGRATION.md [hunk 16]
```diff
@@ -29,3 +30,260 @@ import '../../node_modules/@metamask/design-tokens/dist/styles.css';
...
+- warning-alternative removed
+- warning-disabled removed
+- success-alternative removed
+- success-disabled removed
+- info-alternative removed
+- info-disabled removed
+- network colors removed
+- shadow-primary added
+- shadow-error added
+- button colors removed
```
Comment (georgewrmarshall) [inline MIGRATION.md:289]: These can be removed once they have be cross referenced with the JS and CSS above

File: docs/utils/useJsonColor.ts (hunk 11 above)
Comment (garrettbear) [reply docs/utils/useJsonColor.ts:101]: unfortunately yes and this one is most up to date because it takes into account the white and black colors @georgewrmarshall

File: MIGRATION.md (hunk 16 above)
Comment (garrettbear) [reply MIGRATION.md:289]: @georgewrmarshall you'll need to be a little more specific. I'm not completely understand how you want this written out

File: MIGRATION.md [hunk 17]
```diff
@@ -3,6 +3,7 @@
 This guide provides detailed instructions for migrating your project from one version of the `@metamask/design-tokens` to another.
 
 - [From version 2.1.1 to 3.0.0](#from-version-211-to-300)
+- [From version 3.0.0 to 4.0.0](#from-version-300-to-400)
```
Comment (garrettbear) [inline MIGRATION.md:6]: this wouldn't be a `4.0`, this would be part of `3.0` right? @georgewrmarshall 
```suggestion
- [From version 3.0.0 to 4.0.0](#from-version-300-to-400)
```
```suggestion
```

File: docs/utils/useJsonColor.ts
Comment (garrettbear) [inline docs/utils/useJsonColor.ts:1]: > This file does still currently have some lint errors/warnings but is operational

File: MIGRATION.md (hunk 17 above)
Comment (garrettbear) [reply MIGRATION.md:6]: nevermind i see that v3 already got released

File: src/figma/brandColors.json [hunk 18]
```diff
@@ -608,5 +608,17 @@
...
+    "value": "#ffffff",
+    "type": "color",
+    "parent": "Brand Colors/v1 - current",
+    "description": ""
+  },
+  "black": {
+    "value": "#000000",
+    "type": "color",
+    "parent": "Brand Colors/v1 - current",
+    "description": ""
```
Comment (georgewrmarshall) [inline src/figma/brandColors.json:622]: This may break current storybook

File: MIGRATION.md (hunk 16 above)
Comment (georgewrmarshall) [reply MIGRATION.md:289]: I've listed the specific JS and CSS tokens that engineers should check for. For example, what does "button colors" mean? What do "network colors" refer to? Without these specifics, it's difficult for engineers to understand what to look for when identifying removed design token code. This list should also be used to check for any removed CSS variables.

**Comment by georgewrmarshall** [comment]
Closing in favour of 
- https://github.com/MetaMask/design-tokens/pull/708
- https://github.com/MetaMask/design-tokens/pull/709
